framework/Samplers/CustomSampler.py

```diff
@@ -166,12 +192,10 @@ def localGenerateInput(self,model,myInput):
     """
     # create values dictionary
     for var in self.toBeSampled.keys():
-      self.values[var] = self.pointsToSample[var][self.counter-1]
-    if 'PointProbability' in self.infoFromCustom.keys():
-      self.inputInfo['PointProbability'] = self.infoFromCustom['PointProbability'][self.counter-1]
-    if 'ProbabilityWeight' in self.infoFromCustom.keys():
-      self.inputInfo['ProbabilityWeight'] = self.infoFromCustom['ProbabilityWeight'][self.counter-1]
-    else:
-      self.inputInfo['ProbabilityWeight'] = 1.0
-
+      for subVar in var.split(','):
+        subVar = subVar.strip()
+        self.values[subVar] = self.pointsToSample[subVar][self.counter-1]
+        self.inputInfo['ProbabilityWeight-' + subVar] = self.infoFromCustom['ProbabilityWeight-' + subVar][self.counter-1]
+    self.inputInfo['PointProbability'] = self.infoFromCustom['PointProbability'][self.counter-1]
+    self.inputInfo['ProbabilityWeight'] = self.infoFromCustom['ProbabilityWeight'][self.counter-1]
```

Comment:
I don't see any comments in all of this.  Can we put a comment before each significant set of actions that explains what is happening in human-readable form?  Things like "check requested inputs and outputs are available" and "construct missing probabilities", etc.